datastore/tests/unit/test_client.py

```diff
@@ -934,6 +934,17 @@ def test_transaction_defaults(self):
             self.assertIs(xact, mock_klass.return_value)
             mock_klass.assert_called_once_with(client)
 
+    def test_read_only_transaction_defaults(self):
```

Comment:
Once you've deleted `Client.read_only_transaction()` and added `**kwargs` to `Client.transaction()` you can convert this test to cover the case where you pass `read_only=True`.